### ISSUE TITLE
Bug: [no-explicit-any] suggestion produce invalid rest parameter types

### ISSUE THREAD

**Issue by baevm** [typescript-eslint/typescript-eslint#12813]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.3&showAST=es&fileType=.tsx&code=GYVwdgxgLglg9mABAEwKYQDYEMBOXYIAUAdKbgOYDOAXIlmAJ4CUiA3ogL4BQXEClUOjhxwA7ogC8iEmRxVa9ZpIB8bTgG4eMMFFQ5gWCKkQBhLBmwAjDMdZdE00sQo06jJrQBucGMk0OAW1QoAAs4ZBlnOVdFD0RvX00OIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oA9jGzKdAIZNC6MAG1IiaNA7RIAGkyQAZpW4AVDgFUmAa04B3JuPxxEAXwC64MJZCWgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
function declaration(...args: any) { }

const arrow = (...args: any) => { };

interface Callable {
  (...args: any): void;
  method(...args: any): void;
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-explicit-any": ["error", {"fixToUnknown": true}]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

The rule should not replace `any` with `unknown`. It should either skip the fix entirely or suggest `unknown[]`?

### Actual Result

`any` is replaced with `unknown` producing invalid TypeScript: `TS2370: A rest parameter must be of an array type.`:
```ts
function declaration(...args: unknown) { }

const arrow = (...args: unknown) => { };

interface Callable {
  (...args: unknown): void;
  method(...args: unknown): void;
}
```

**Comment by kirkwaiblinger** [comment]
Seems reasonable to use `unknown[]` in the case of rest args 👍
